Logtail.NLog/LogtailTarget.cs

```diff
@@ -45,24 +43,54 @@ public sealed class LogtailTarget : TargetWithContext, IUsesStackTrace
         /// We capture the file and line of every log message by default. You can turn this
         /// option off if it has negative impact on the performance of your application.
         /// </summary>
-        public bool CaptureSourceLocation { get; set; } = true;
+        public bool CaptureSourceLocation
+        {
+            get => StackTraceUsage == StackTraceUsage.Max;
+            set => StackTraceUsage = value ? StackTraceUsage.Max : StackTraceUsage.WithoutSource;
+        }
 
         /// <summary>
         /// Include GlobalDiagnosticContext in logs.
         /// </summary>
         public bool IncludeGlobalDiagnosticContext { get; set; } = true;
 
-        public StackTraceUsage StackTraceUsage { get; private set; } = StackTraceUsage.Max;
+        public StackTraceUsage StackTraceUsage
+        {
+            get => _stackTraceUsage ?? StackTraceUsage.Max;
+            set
+            {
+                if (value == StackTraceUsage.None)
+                {
+                    IncludeCallSite = false;
+                    IncludeCallSiteStackTrace = false;
+                }
+                else
+                {
+                    IncludeCallSite = true;
+                    IncludeCallSiteStackTrace = value == StackTraceUsage.Max;
+                }
+                _stackTraceUsage = value;
+            }
+        }
+        private StackTraceUsage? _stackTraceUsage = null;
```

Comment:
This effectively turns this off by default, while it was turned on before. Would you be able to rewrite this such the default is turned on?

Reply:
Have now updated the source-code, so it is more obvious that it is `StackTraceUsage.Max` by default.